### PR TITLE
Allow updating npm lockfile v2 with the new version

### PR DESCRIPTION
**Pull request by havenchyk**
Hey @phips28 

This is a draft PR and I didn't even test it, but can you help me with testing it?

# Description

re #81 

The aim of this PR is to support updating `package-lock.json` together with the `package.json`

`lockfileVesion: 2` adds the info about the version in `package-lock.json` - to check it out please try to run `npm i` with npm@7 in this repo, you'll get something like this 

img

Package version in package.json and package-lock.json is different.

The goal of this PR is to update version in package-lock.json together with the version in package.json.
Basically you can run npm script that doesn't compare node_modules with package-lock.json and updates package-lock.json with the new version from package.json

### CODE REVIEW

**Comment by phips28** [comment]
Can you add a proper description? Thx

**Comment by havenchyk** [comment]
hey @phips28 updated!

File: index.js [hunk 1]
```diff
@@ -157,6 +157,17 @@ const workspace = process.env.GITHUB_WORKSPACE;
     newVersion = execSync(`npm version --git-tag-version=false ${version}`).toString().trim().replace(/^v/, '');
     newVersion = `${tagPrefix}${newVersion}`;
     console.log(`::set-output name=newTag::${newVersion}`);
+
+    // case: when user wants to skip updating package-lock.json v2
+    try {
+      const bumpPackageLock = process.env['INPUT_BUMP_PACKAGE-LOCK']
+      if (bumpPackageLock) {
```
Comment (phips28) [inline index.js:164]: must be `bumpPackageLock === 'true'` as it is a string.

**Review by phips28 — CHANGES_REQUESTED** [review]

**Review by phips28 — APPROVED** [review]

**Comment by phips28** [comment]
@havenchyk I tested this in production but I had to revert it. Because with the new version, where we use node12, this would downgrade the lockfile from v2 to v1. And this is not what we want.

**Comment by havenchyk** [comment]
Thanks for checking out. What do you think if we disable this behaviour by default and enable it only via flag with the proper message? I really don't want to amend a commit from your action

**Comment by phips28** [comment]
its not possible as google actions does not have node14 with npm7 support. But we need npm7 to keep lockfile v2.

This would have worked with v8 (docker) - but we switched to node12 only action to increase speed from 1min to a few secs.
